Training/Model/NoteRepository.php

```diff
@@ -0,0 +1,162 @@
+ */
+
+declare(strict_types=1);
+
+namespace Codifi\Training\Model;
+
+use Codifi\Training\Api\Data\NoteInterface;
+use Codifi\Training\Api\NoteRepositoryInterface;
+use Codifi\Training\Model\CustomerNote;
+use Codifi\Training\Model\CustomerNoteFactory;
+use Codifi\Training\Model\ResourceModel\CustomerNote as CustomerNoteResourse;
+use Codifi\Training\Api\Data\NoteSearchResultInterfaceFactory;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Exception\AlreadyExistsException;
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Codifi\Training\Api\Data\NoteSearchResultInterface;
+use Magento\Framework\Api\SearchCriteria\CollectionProcessorInterface;
+
+/**
+ * Class NoteRepository
+ * @package Codifi\Training\Model
+ */
+class NoteRepository implements NoteRepositoryInterface
+{
+    /**
+     * Note interface.
+     *
+     * @var NoteInterface
+     */
+    private $note;
+
+    /**
+     * Note factory.
+     *
+     * @var CustomerNoteFactory
+     */
+    private $noteFactory;
+
+    /**
+     * Note resourse.
+     *
+     * @var CustomerNoteResourse
+     */
+    private $noteResourse;
+
+    /**
+     * @var NoteSearchResultInterfaceFactory
+     */
+    private $searchResultFactory;
+
+    /**
+     * @var CollectionProcessorInterface
+     */
+    private $collectionProcessor;
+
+    /**
+     * NoteRepository constructor.
+     *
+     * @param NoteInterface $note
+     * @param CustomerNoteFactory $noteFactory
+     * @param CustomerNoteResourse $noteResourse
+     * @param NoteSearchResultInterfaceFactory $searchResultFactory
+     * @param CollectionProcessorInterface $collectionProcessor
+     */
+    public function __construct(
+        NoteInterface $note,
+        CustomerNoteFactory $noteFactory,
+        CustomerNoteResourse $noteResourse,
+        NoteSearchResultInterfaceFactory $searchResultFactory,
+        CollectionProcessorInterface $collectionProcessor
+    ) {
+        $this->note = $note;
+        $this->noteFactory = $noteFactory;
+        $this->noteResourse = $noteResourse;
+        $this->searchResultFactory = $searchResultFactory;
+        $this->collectionProcessor = $collectionProcessor;
+    }
+
+    /**
+     * Get note by id.
+     *
+     * @param int $id
+     * @return CustomerNote
+     * @throws NoSuchEntityException
+     */
+    public function getById($id): CustomerNote
+    {
+        $noteModel = $this->noteFactory->create();
+        $this->noteResourse->load($noteModel, $id);
+        if (!$noteModel->getId()) {
+            throw new NoSuchEntityException(__('Unable to find note with ID "%1"', $id));
+        }
+
+        return $noteModel;
+    }
+
+    /**
+     * Save note.
+     *
+     * @param NoteInterface $note
+     * @throws AlreadyExistsException
+     */
+    public function save(NoteInterface $note): void
+    {
+        $noteModel = $this->noteFactory->create();
+        $noteModel->setData($note);
```

Comment:
1. See Magento\Framework\Model\AbstractModel::setData
2. You don't need to create model for save it
3. Don't forget to handle exceptions